pkg/controllers/state/statenode.go

```diff
@@ -135,6 +135,16 @@ func (in *StateNode) Name() string {
 	return in.Node.Name
 }
 
+// ProviderID is the key that is used to map this StateNode
+// If the Node and NodeClaim have a providerID, this should map to a real providerID
+// If the Node does not have a providerID, this will map to the node name
```

Comment:
How does this map to the node name? Is this a typo, because I don't see a reference to the  node name in the code.

Reply:
Here's the code pointer: https://github.com/aws/karpenter-core/blob/main/pkg/controllers/state/cluster.go#L239